### PR TITLE
Remove elrond-go dependency feat/new-indexer

### PR DESCRIPTION
**Pull request by miiu96**
Use the latest commit from **elrond-go** repository development branch

- removed elrond-go and arwen dependencies 
- removed tps benchmark indexer and the implementation
- removed logic that save validators public keys from indexer ( moved in elrond-go)

### CODE REVIEW

File: mock/userAccountStub.go [hunk 1]
```diff
@@ -2,42 +2,18 @@ package mock
...
-	if u.AddToBalanceCalled != nil {
-		return u.AddToBalanceCalled(value)
-	}
-	return nil
-}
-
-// SubFromBalance -
-func (u *UserAccountStub) SubFromBalance(_ *big.Int) error {
-	return nil
+func (u *UserAccountStub) IncreaseNonce(_ uint64) {
```
Comment (iulianpascalau) [inline mock/userAccountStub.go:15]: comments

File: mock/userAccountStub.go [hunk 2]
```diff
@@ -2,42 +2,18 @@ package mock
...
+	panic("implement me")
```
Comment (iulianpascalau) [inline mock/userAccountStub.go:16]: please make a proper stub

File: process/transactions/scDeploysProcessor.go [hunk 3]
```diff
@@ -6,12 +6,13 @@ import (
...
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/hashing/keccak"
-	"github.com/ElrondNetwork/elrond-go/process/factory"
-	"github.com/ElrondNetwork/elrond-vm-common/parsers"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/hashing/keccak"
 )
 
+// ArwenVirtualMachine is a byte array identifier for the smart contract address created for Arwen VM
+var ArwenVirtualMachine = []byte{5, 0}
```
Comment (iulianpascalau) [inline process/transactions/scDeploysProcessor.go:14]: move to core, please

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: mock/userAccountStub.go (hunk 1 above)
Comment (miiu96) [reply mock/userAccountStub.go:15]: done.

File: mock/userAccountStub.go (hunk 2 above)
Comment (miiu96) [reply mock/userAccountStub.go:16]: done.

File: process/transactions/scDeploysProcessor.go (hunk 3 above)
Comment (miiu96) [reply process/transactions/scDeploysProcessor.go:14]: This const will be removed in the next PR where sc deploys processor was refactored.

File: go.mod [hunk 4]
```diff
@@ -3,11 +3,9 @@ module github.com/ElrondNetwork/elastic-indexer-go
 go 1.15
 
 require (
-	github.com/ElrondNetwork/elrond-go v1.2.4-0.20210701140353-fb591fb0f9ae
-	github.com/ElrondNetwork/elrond-go-logger v1.0.4
-	github.com/ElrondNetwork/elrond-vm-common v0.3.4-0.20210625150635-525789454b6f
+	github.com/ElrondNetwork/elrond-go-core v1.0.1-0.20210722103906-9936a73f9543
```
Comment (bogdan-rosianu) [inline go.mod:6]: :tada:

File: interface.go [hunk 5]
```diff
@@ -38,7 +36,34 @@ type ElasticProcessor interface {
...
-	ComputeTxFeeBasedOnGasUsed(tx process.TransactionWithFeeHandler, gasUsed uint64) *big.Int
-	ComputeMoveBalanceGasUsed(tx process.TransactionWithFeeHandler) uint64
+	ComputeGasUsedAndFeeBasedOnRefundValue(tx coreData.TransactionWithFeeHandler, refundValue *big.Int) (uint64, *big.Int)
+	ComputeTxFeeBasedOnGasUsed(tx coreData.TransactionWithFeeHandler, gasUsed uint64) *big.Int
+	ComputeGasLimit(tx coreData.TransactionWithFeeHandler) uint64
+	IsInterfaceNil() bool
+}
+
+// Coordinator defines what a shard state coordinator should hold
+type Coordinator interface {
```
Comment (bogdan-rosianu) [inline interface.go:46]: please rename to `ShardCoordinator` as we also have `NodesCoordinator`
Before, it was accessed by `sharding.Coordinator` which made sense. But now, it doesn't

File: mock/userAccountStub.go [hunk 6]
```diff
@@ -48,32 +24,12 @@ func (u *UserAccountStub) GetBalance() *big.Int {
...
-
-// ChangeOwnerAddress -
-func (u *UserAccountStub) ChangeOwnerAddress([]byte, []byte) error {
-	return nil
-}
-
-// SetOwnerAddress -
-func (u *UserAccountStub) SetOwnerAddress([]byte) {
-}
+func (u *UserAccountStub) RetrieveValueFromDataTrieTracker(key []byte) ([]byte, error) {
```
Comment (bogdan-rosianu) [inline mock/userAccountStub.go:27]: comment

File: process/statistics/serialize.go [hunk 7]
```diff
@@ -6,73 +6,17 @@ import (
...
-
-	for _, shardInfo := range shardsInfo {
-		errSerialize := serializeShardInfo(buff, shardInfo, index)
-		if errSerialize != nil {
-			log.Warn("serializeShardInfo", "shardID", shardInfo.ShardID, "error", errSerialize)
-
-			continue
-		}
-	}
+var log = logger.GetOrCreate("process/statistics")
```
Comment (bogdan-rosianu) [inline process/statistics/serialize.go:12]: in some places you use `logger.GetOrCreate("indexer/<path>")` and here `logger.GetOrCreate("<path>")`
please refactor in the entire project so you use one of them. I preferred the one with `indexer` preffix

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (miiu96) [reply go.mod:6]: done.

File: interface.go (hunk 5 above)
Comment (miiu96) [reply interface.go:46]: done.

File: mock/userAccountStub.go (hunk 6 above)
Comment (miiu96) [reply mock/userAccountStub.go:27]: done.

File: process/statistics/serialize.go (hunk 7 above)
Comment (miiu96) [reply process/statistics/serialize.go:12]: done. gj

**Review by bogdan-rosianu — APPROVED** [review]

File: process/transactions/scDeploysProcessor.go (hunk 3 above)
Comment (iulianpascalau) [reply process/transactions/scDeploysProcessor.go:14]: great

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**